File: 2016/gara1/estintori/att/estintori.pas
const
    MAXR = 1000;
    MAXC = 1000;
type matrix = array[0..MAXR] of array[0..MAXC] of char;

function mancanti(R, C: longint; var M: matrix): longint;
begin
    (* Mettete qui il codice della soluzione *)
    mancanti := 42;
end;


var
    R, C, i: longint;
    M: array[0..MAXR] of array[0..MAXC] of char;
    fr, fw: text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, R, C);
    for i:=0 to R-1 do
        readln(fr, M[i]);

    writeln(fw, mancanti(R, C, M));
    close(fr);
    close(fw);
end.

File: 2016/gara1/estintori/att/estintori.c
#include <stdio.h>
#include <stdlib.h>
#include <assert.h>


int mancanti(int R, int C, char** map) {
    // Mettete qui il codice della soluzione
    return 42;
}


int main() {
    FILE *fr, *fw;
    int R, C, i;
    char** map;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d%d", &R, &C));
    map = (char**) malloc(R * sizeof(char*));
    for(i=0; i<R; i++) {
        map[i] = (char*) malloc(C * sizeof(char*));
        assert(1 == fscanf(fr, "%s", map[i]));
    }

    fprintf(fw, "%d\n", mancanti(R, C, map));
    fclose(fr);
    fclose(fw);
    return 0;
}
